### ISSUE TITLE
Backward replication of RECAP PDF uploads to subdockets

### ISSUE THREAD

**Issue by albertisfu** [freelawproject/courtlistener#4864]
This is a follow up of #4826

In https://github.com/freelawproject/courtlistener/pull/4857 we added support for replicating incoming PDF uploads to subdockets. The goal of this issue is to enable PDF replication in the opposite direction. Specifically, if a docket already contains multiple merged PDFs and we detect that it has subdockets, we should replicate the PDFs to all subdockets where the `RECAPDocuments` match. This ensures that each subdocket is as complete as possible.

### To consider:

- **Replication at the `RECAPDocument` Level:**
    
    PDFs should be copied at the `RECAPDocument` level, matched by `pacer_doc_id` in order to avoid merging content that does not belong to a sub-docket.
    
- **Triggering Backward Replication:**
    
    What action should trigger this backward replication? Ideally, replication should occur as soon as we identify that a docket has subdockets. Currently, this can be detected during an attachment page upload or a PDF upload, where logic exists to find common `RECAPDocuments` in subdockets. Additional triggers may be required once we add this logic to other sources.
    
- **Handling content variations in subdockets:**
    
    Some sub-dockets may have different PDFs than others. To ensure completeness, the replication logic should be applied to each subdocket to all other sub-dockets. By the end of the process, all subdockets will contain the all the same content.

**Comment by mlissner** [comment]
I might not be thinking about this carefully enough, but I think the trigger is that whenever you add a new RECAPDocument without a PDF, you check if the PDF is available elsewhere and you replicate it to the new RECAPDocument if needed. 

I think this happens when:
 - We get new docket entries from a docket upload, docket history upload, or recap.email email. 
 - We get new docket entries from RSS
 - We get new docket entries from an attachment page.

We could have an extra step to only do this if the docket number contains `cr`, to save on performance, but otherwise, seems straightforward to me?

**Comment by albertisfu** [comment]
Got it! So the goal of this issue is simply to replicate existing PDFs to new RECAPDocuments that match, correct?

I initially thought we wanted to replicate existing PDFs to all RECAPDocuments that matched, even if the RDs were not new. However, if this should only apply to newly incoming RECAPDocuments, I agree that it’s straightforward.

**Comment by mlissner** [comment]
> Got it! So the goal of this issue is simply to replicate existing PDFs to new RECAPDocuments that match, correct?

Yes, exactly. We'll get to that bigger project eventually, but not now. :)
